test/hw4/test_task3.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from All_home_works.hw4.task3 import my_precious_logger
+
+
+def test_my_precious_logger__error_msg__test_msg_in_stderr(capsys):
+    error = "error: 404"
+    my_precious_logger(text=error)
+    captured = capsys.readouterr()
+    assert captured.err == error
+    assert not captured.out
+
+
+@pytest.mark.parametrize("input_data", ["Error: 404", "error 404"])
```

Comment:
Almost there, need to find one more combination with `error`.  There is a keyword for it in the requirements. 

Also, this test helped me to find a bug in your code.  One of the `input_data` should be in stderr.

Reply:
Task description:

> if line starts with "error"
My mistake is: 'error:'

And next combination is: 'errorTEXT' or 'errorerror'?

Reply:
We are not considering bytes string? I do not know.

Reply:
> We are not considering bytes string? 

This looks like a randomly dropped suggestion.  I don't see any hints in the phrase `if a line starts with "error"` which could lead to bytes string.   

There is a technique named brainstorming when you throw some raw ideas. But brainstorming is not used in the current context. In teamwork, the fast and random answer gives more problems than it solves. Just simple `I do not know` would be much better. 

You need to find some cases that could violate this requirement:
```
if line starts with "error"
```

Reply:
I'm sorry, but I do not know...

Reply:
if "error" in line, but not line startswith "error"

Reply:
Exactly!